tests/degradation_test_v1.py

```diff
@@ -0,0 +1,182 @@
+        elif (input_freq == 'D'):
+            x = pd.date_range(start='2012-01-01', end='2015-01-01', freq=input_freq)
+            N = len(x)
+            days = np.arange(N)
+            y = np.ones(N) * np.power(1 + rd/365, days)
+        else :
+            sys.exit("Unknown frequency type")
+
+        corr_energy = pd.Series(data=y, index=x)
+        return corr_energy
+
+    def setUp(self):
+        # define module constants and parameters
+
+        self.rd = -0.005
+
+        if self.input_freq in self.list_CD_YOY_input_freq:
+            self.test_CD_YOY_corr_energy = self.get_CD_YOY_corr_energy(self.rd, self.input_freq)
+        
+        self.test_ols_corr_energy = self.get_ols_corr_energy(self.rd, self.input_freq)
+             
+
+    def tearDown(self):
+        pass
+    
+    def test_degradation_with_ols(self):
+        ''' Test degradation with ols. '''
+
+        funcName = sys._getframe().f_code.co_name
+        print '\r', 'Running ', funcName
+
+        # test ols degradation calc
+        print 'Frequency: ', self.input_freq
+        rd_result = degradation_ols(self.test_ols_corr_energy)
+        self.assertAlmostEqual(rd_result[0], 100*self.rd, places=1)
+        print 'Actual: ', 100*self.rd
+        print 'Estimated: ', rd_result[0]
+
+        # TODO
+        # - support for different time series frequencies
+        # - inputs
+
+    def test_degradation_classical_decomposition(self):
+        ''' Test degradation with classical decomposition. '''
+
+        funcName = sys._getframe().f_code.co_name
+        print '\r', 'Running ', funcName
+
+        # test classical decomposition degradation calc
+        print 'Frequency: ', self.input_freq
+        rd_result = degradation_classical_decomposition(self.test_CD_YOY_corr_energy)
+        self.assertAlmostEqual(rd_result[0], 100*self.rd, places=1)
+        print 'Actual: ', 100*self.rd
+        print 'Estimated: ', rd_result[0]
+
+        # TODO
+        # - support for different time series frequencies
+        # - inputs
+
+    def test_degradation_year_on_year(self):
+        ''' Test degradation with year on year approach. '''
+
+        funcName = sys._getframe().f_code.co_name
+        print '\r', 'Running ', funcName
+
+        # test classical decomposition degradation calc
+        print 'Frequency: ', self.input_freq
+        rd_result = degradation_year_on_year(self.test_CD_YOY_corr_energy)
+        self.assertAlmostEqual(rd_result[0], 100*self.rd, places=1)
+        print 'Actual: ', 100*self.rd
+        print 'Estimated: ', rd_result[0]
+
+        # TODO
+        # - support for different time series frequencies
+        # - inputs
+
+
+if __name__ == '__main__':
+
+    # Allowed frequencies for degradation_ols
+    list_ols_input_freq = ['MS','M','W','D','H','T','S']
+    # Allowed frequencies for degradation_classical_decomposition and degradation_year_on_year
+    list_CD_YOY_input_freq = ['MS','M','W','D']
```

Comment:
I prefer v2 to v1. Either way gets the job done, but it is better practice to define variables for the tests inside the test classes themselves.